code/datums/emotes.dm

```diff
@@ -0,0 +1,169 @@
+#define EMOTE_VISIBLE 1
+#define EMOTE_AUDIBLE 2
+
+/datum/emote
+	var/key = "" //What calls the emote
+	var/key_third_person = "" //This will also call the emote
+	var/message = "" //Message displayed when emote is used
+	var/message_alien = "" //Message displayed if the user is a grown alien
+	var/message_larva = "" //Message displayed if the user is an alien larva
+	var/message_robot = "" //Message displayed if the user is a robot
+	var/message_AI = "" //Message displayed if the user is an AI
+	var/message_monkey = "" //Message displayed if the user is a monkey
+	var/message_simple = "" //Message to display if the user is a simple_animal
+	var/message_param = "" //Message to display if a param was given
+	var/emote_type = EMOTE_VISIBLE //Whether the emote is visible or audible
+	var/restraint_check = FALSE //Checks if the mob is restrained before performing the emote
+	var/muzzle_ignore = FALSE //Will only work if the emote is EMOTE_AUDIBLE
+	var/list/mob_type_allowed_typecache = /mob //Types that are allowed to use that emote
+	var/list/mob_type_blacklist_typecache //Types that are NOT allowed to use that emote
+	var/list/mob_type_ignore_stat_typecache
+	var/stat_allowed = CONSCIOUS
```

Comment:
Can this and restraint check be refactored into bitflags?

Reply:
Again, out of scope.

Reply:
in-scope imo

Reply:
`stat_allowed` conversion to bitflags wouldn't be that clean, it's not a boolean value.
`restraint_check`, `muzzle_ignore`, `vary` and `only_forced_audio` would make more sense.

Reply:
1. Losing compatibility with /tg/
2. Look at the previous emote code. This is 10000x better.

Reply:
Though I'll rather pay this ransom than let this PR sit for a week since it's actively hurting me. You win again.

Reply:
> 
> 
> Can this and restraint check be refactored into bitflags?

stat is not a bitflag